### PR TITLE
Add Tagger Script support to the code highlighter.

### PR DESCRIPTION
**Pull request by Gremious**
Added support for the [Tagger Script](https://picard.musicbrainz.org/docs/scripting/) language in the code highlighter.

### CODE REVIEW

File: markdownhighlighter.cpp [hunk 1]
```diff
@@ -493,7 +493,8 @@ void MarkdownHighlighter::initCodeLangs()
         {QLatin1String("v"),           MarkdownHighlighter::CodeV},
         {QLatin1String("xml"),         MarkdownHighlighter::CodeXML},
         {QLatin1String("yml"),         MarkdownHighlighter::CodeYAML},
-        {QLatin1String("yaml"),        MarkdownHighlighter::CodeYAML}
+        {QLatin1String("yaml"),        MarkdownHighlighter::CodeYAML},
+        {QLatin1String("taggerscript"),MarkdownHighlighter::CodeTaggerScript}
```
Comment (Waqar144) [inline markdownhighlighter.cpp]: Could you move it up? (in alphabetical order)

File: markdownhighlighter.cpp [hunk 2]
```diff
@@ -1227,6 +1231,60 @@ int MarkdownHighlighter::highlightNumericLiterals(const QString &text, int i)
...
+        if (text.at(i) == QChar('$') && text.midRef(i, 5) != QLatin1String("$noop") ) {
+            int next = text.indexOf(QChar('('), i);
+            if (next == -1) break;
+            setFormat(i, next-i, _formats[CodeKeyWord]);
+            i = next;
+        }
+
+        //highlight variables
+        if (text.at(i) == QChar('%')) {
+            int next = text.indexOf(QChar('%'), i+1);
```
Comment (Waqar144) [inline markdownhighlighter.cpp]: do `++i` before this, so that you don't have to manually add +1

**Comment by Waqar144** [comment]
Do you have any example codes available for testing?

**Comment by Gremious** [comment]
You can find exmaple codes [here](https://picard.musicbrainz.org/docs/scripting/#coverart_naming_examples)

Alternatively, my messy test code:
```taggerscript
%asd%%asd%%asd%%%%$noop(asdsa)%%%%%%%%%%%%%% -repeated highlighting and error
$noop(comment)

$if(if,  %then%, Cool Escape Sequence \wow \"super" \NICE!!) 

nested highlighting, first declared takes priority:
$noop%asd(asdsa$)asd(
$aaaa$noop(asdsa)(%
%something $else% (%

escaping works as expected
doesn't break if it's just on it's own as last char
\\(asd
\$asdasd(asd)
\%asd%
\|""asd"sad
\
```

**Comment by pbek** [comment]
## 20.1.2
- added tagger script code block highlighting support (for [#68](https://github.com/pbek/qmarkdowntextedit/pull/68), thank you @Gremious)

There now is a new release, could you please test it and report if it works for you?

**Pull request merged**